### DISCUSSION TITLE
Keycloak Roadmap for 2023

### DISCUSSION THREAD

**Discussion by SentinelCQ** [keycloak/keycloak#17096] [Q&A]
I have been using this page to track the releases for Keycloak.

https://www.keycloak.org/2022/03/releases.html

But it is about to reach its end since it only listed plans out to Keycloak 22. Interested to see what is the plan for 2023, and what the updated timeline is for Keycloak 21 and 22. 21 has an original release date of Dec/Jan 2023 and I have not yet seen an announcement or a branch.

**Comment by jonkoops** [comment]
Yeah, it might be time for a new roadmap for this year. WDYT @stianst?

**Reply by stianst** [reply]
Already on my to-do list, but yeah it's well overdue.

**Reply by SentinelCQ** [reply]
Thanks. Is there an updated timeline for Keycloak 21?

**Reply by StefanOverHaevgRZ** [reply]
@stianst What's the major driver for the roadmap? Is it rather driven by Red Hats internal ambitions, or by the community requests?

**Reply by stianst** [reply]
> @stianst What's the major driver for the roadmap? Is it rather driven by Red Hats internal ambitions, or by the community requests?

We take everything into consideration and work on what we believe is most important and brings the most value to both community and Red Hat

**Comment by SentinelCQ** [comment]
Any updates on the roadmap? Just curious since there seems to be a shift in the release cadence/numbering to the 2nd digit releases 21.1 vs 2022 where it was far more first digit releases with the ocassional 3rd digit/patch release.

**Reply by vmuzikar** [reply]
The next version will be 22.0.0, 21.1.z was a one-off. ;)

**Reply by jasonhjia** [reply]
when will 22.0.0 be released? I see code difference between main branch and 21.x branch

**Reply by jonkoops** [reply]
@jasonhjia there is already a discussion for that (see #20278)

**Comment by Paul6552** [comment]
Maybe we can have a roadmap for 2024? 2023 is almost over x)

I would be interested to see where the journey with keycloak takes us

**Reply by tgau** [reply]
Any update for 2024/25?
